reposerver/repository/repository.go

```diff
@@ -1517,7 +1517,7 @@ func GenerateManifests(ctx context.Context, appPath, repoRoot, revision string,
 		// if pluginName is provided it has to be `<metadata.name>-<spec.version>` or just `<metadata.name>` if plugin version is empty
 		targetObjs, err = runConfigManagementPluginSidecars(ctx, appPath, repoRoot, pluginName, env, q, q.Repo.GetGitCreds(gitCredsStore), opt.cmpTarDoneCh, opt.cmpTarExcludedGlobs, opt.cmpUseManifestGeneratePaths)
 		if err != nil {
-			err = fmt.Errorf("plugin sidecar failed. %s", err.Error())
+			err = fmt.Errorf("could not find CMP sidecar for application %q: %s", q.AppName, err.Error())
```

Comment:
`runConfigManagementPluginSidecar` can return a lot more error, than just not finding the CMP sidecar. While the old error message, isn't really helpful, the new one is misleading which is worse.
For example error like this ( https://github.com/argoproj/argo-cd/blob/c9fd4409168b8f1cf1341409a15c1a6ef50c84c3/reposerver/repository/repository.go#L2116 ) would be reported as `could no find CMP sidecar`

To improve on this message here, it would need to differentiate between the different error messages.